EDU-18977: add offsite capture web script installation guide (#2862)
vtexdocs/dev-portal-content: docs/guides/VTEX Ads/offsite-capture/installing-the-offsite-capture-web-script.md

---
title: "Installing the offsite capture web script"
slug: "installing-the-offsite-capture-web-script"
hidden: false
createdAt: "2026-07-10T00:00:00.000Z"
updatedAt: "2026-07-10T00:00:00.000Z"
excerpt: "Install and configure the VTEX Ads offsite capture web script on VTEX native or independent storefronts so offsite clicks and conversions are attributed to your campaigns."
---

> Offsite traffic and conversion tracking rely on a lightweight **VTEX Ads** web script that runs on your storefront. This guide explains how offsite capture works and walks you through publisher provisioning, installing the script for your store type, validating capture, and troubleshooting zeroed metrics.

Reviewer: Não sei se "zeroed" foi uma decisão técnica, mas esse termo costuma significar que o valor foi deliberadamente definido como zero ("zeroed" == "zerado"). Não significa que ele venha assim porque não houve registro ou que haja algum problema.
```suggestion
Offsite traffic and conversion tracking rely on a lightweight **VTEX Ads** web script that runs on your storefront. This guide explains how offsite capture works and walks you through publisher provisioning, installing the script for your store type, validating capture, and troubleshooting metrics showing 0.
```

- You know which installation path applies to your storefront (VTEX native or independent). Use the table below.

| Your store                                             | How the script is added                                                          | Who installs       |
| ------------------------------------------------------ | -------------------------------------------------------------------------------- | ------------------ |
| VTEX native store (Store Framework, Portal, FastStore) | Potentially installed as a VTEX app (confirm eligibility with the VTEX Ads team) | VTEX / store admin |
| Independent store (not built on the VTEX storefront)   | A JavaScript snippet loaded from the VTEX Ads CDN                                | Your team          |

> ℹ️ VTEX native store eligibility for the VTEX Ads Agent app may vary. Confirm with the VTEX Ads team before you install.

## How offsite capture works

Offsite capture connects an ad click to a later purchase on your storefront. The flow below summarizes how attribution is established.

> 1. The offsite ad's destination URL carries the **VTEX Ads** annotators (URL parameters that preserve campaign and click attribution data). On VTEX native stores, the click may also pass through the intermediate URL (`/va/go`), which preserves the annotators until the user reaches the destination retailer.

Reviewer: Sugestão do Grammarly de melhoria de escrita.
```suggestion
1. The offsite ad's destination URL carries the **VTEX Ads** annotators (URL parameters that preserve campaign and click attribution data). In VTEX native stores, the click may also pass through the intermediate URL (`/va/go`), preserving the annotators until the user reaches the destination retailer.
```

2. When the user lands on the destination retailer, the web script captures the session and the page view.
3. **VTEX Ads** matches that session or journey to the offsite click and, when an order is placed, attributes the conversion to the campaign. Attribution is subject to the applicable conversion window.

For how sessions, deduplication, and conversion windows are defined in **VTEX Ads**, see [Understanding ads events](https://developers.vtex.com/docs/guides/understanding-ads-events).

### Requirements for capture to work

Capture succeeds only when all of the following are true:

- The offsite media is served with the **VTEX Ads** parameters preserved.
- The redirect to the destination URL provided by **VTEX Ads** is preserved (not shortened or altered).
- The destination page actually loads for the user. Capture happens on load.

For journeys that finish inside a mobile app, the [Activity Flow Mobile SDK](https://developers.vtex.com/docs/guides/installing-activity-flow-in-mobile-apps) is also required. See [Activity Flow](https://developers.vtex.com/docs/guides/activity-flow) for an overview of the SDK and supported platforms.

## 1. Request publisher provisioning

> To run offsite campaigns that drive traffic to your store, your store must first be registered in **VTEX Ads** as a publisher. This setup is handled by the VTEX Ads team. You do not create the publisher account yourself.

Reviewer: Mudar de voz passiva para ativa.
```suggestion
To run offsite campaigns that drive traffic to your store, your store must first be registered in **VTEX Ads** as a publisher. The VTEX Ads team handles this setup. You do not create the publisher account yourself.
```

Install the app with the VTEX IO CLI:

```bash
vtex install vtex.ads-agent
```

After installing, set `publisherId` in the app settings so the script starts reporting events. For full install and configuration steps, see the [`vtex.ads-agent` app documentation](https://developers.vtex.com/docs/apps/vtex.ads-agent).

## 3. Validate your installation

Follow these checks after installing the script:

1. Access the destination retailer through an offsite URL (the parameterized URL generated for the campaign).
2. Confirm the script loads and that the offsite access event is sent. Check the network request.
3. Run a control case: an access to the same page **without** the offsite parameters should **not** be registered as an offsite click.
4. Run an end-to-end test: with a test offsite campaign, place a test order and confirm that all metrics (impression, click, and conversion) were captured. This end-to-end validation is the retailer's responsibility.

> ℹ️ The expected network endpoint, request name, or payload signature for offsite access events has not been documented here. Confirm what to look for in DevTools with the VTEX Ads team if validation is unclear.

## Troubleshooting

> ### Zeroed metrics

Reviewer: ```suggestion
### Metrics remain at 0
```

> If the script is not loading on the destination pages, there is no capture and metrics come out zeroed. Confirm the script tag is present on the destination pages and that the bundle loads without errors.

Reviewer: ```suggestion
If the script is not loading on the destination pages, there is no capture, and metrics remain at 0. Confirm the script tag is present on the destination pages and that the bundle loads without errors.
```
